cocotb/handle.py

```diff
@@ -291,14 +291,15 @@ def __init__(self, handle, handle_type):
             self._value = simulator.get_signal_val_real(self._handle)
         else:
             self._value = BinaryValue()
-            self._value.binstr = simulator.get_signal_val_str(self._handle)
+            try:
+                self._value.binstr = simulator.get_signal_val_str(self._handle)
```

Comment:
Can you get the value from sim and then try and assign to Binary and then try to _value rather than go down to sim and get it again.